src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/armor/SlimefunArmorPiece.java

```diff
@@ -29,4 +37,12 @@ public SlimefunArmorPiece(Category category, SlimefunItemStack item, RecipeType
         return effects;
     }
 
+    /**
+     * This returns the armor set {@link NamespacedKey} of this {@link SlimefunArmorPiece}.
+     *
+     * @return The set {@link NamespacedKey}, <code>null</code> if none is found.
+     */
+    public NamespacedKey getSetId() {
+        return id;
+    }
```

Comment:
This method should probably be moved to your `CustomProtection` interface and then implemented in the Hazmat class.